src/com/owncloud/android/lib/common/utils/Log_OC.java

```diff
@@ -73,30 +74,26 @@ public static void wtf(String TAG, String message) {
 
     /**
      * Start doing logging
-     * @param storagePath : directory for keeping logs
+     * @param context Context: used for determinated app specific folder
      */
-    synchronized public static void startLogging(String storagePath) {
-        String logPath = storagePath + File.separator +
-                mNextcloudDataFolderLog + File.separator + LOG_FOLDER_NAME;
-        mFolder = new File(logPath);
+    synchronized public static void startLogging(Context context) {
+        mNextcloudDataFolderLog = context.getFilesDir().getAbsolutePath() + File.separator + LOG_FOLDER_NAME;
+        mFolder = new File(mNextcloudDataFolderLog);
         mLogFile = new File(mFolder + File.separator + mLogFileNames[0]);
 
-        boolean isFileCreated = false;
+        boolean isFolderCreated = false;
 
         if (!mFolder.exists()) {
-            mFolder.mkdirs();
-            isFileCreated = true;
-            Log.d(TAG, "Log file created");
+            isFolderCreated = mFolder.mkdirs();
+            Log.d(TAG, "Log folder created at: " + mNextcloudDataFolderLog);
         }
 
-        try { 
-
+        try {
             // Create the current log file if does not exist
-            mLogFile.createNewFile();
+            isEnabled = mLogFile.createNewFile();
             mBuf = new BufferedWriter(new FileWriter(mLogFile, true));
-            isEnabled = true;
 
-            if (isFileCreated) {
+            if (isFolderCreated && isEnabled) {
```

Comment:
Should not happen as this is the app specific folder which will be deleted when the app is removed.
But I have added it, just to be safe.